nomenclature/processor/generic.py

```diff
@@ -0,0 +1,153 @@
+import logging
+from collections import Counter
+from pathlib import Path
+
+import yaml
+from pyam import IamDataFrame
+from pydantic import BaseModel, field_validator, ValidationInfo
+from pydantic.types import FilePath
+from pydantic_core import PydanticCustomError
+
+from nomenclature.definition import DataStructureDefinition
+from nomenclature.error import custom_pydantic_errors
+from nomenclature.processor import Processor
+from nomenclature.processor.utils import get_relative_path
+
+logger = logging.getLogger(__name__)
+
+here = Path(__file__).parent.absolute()
+
+
+class AggregationItem(BaseModel):
```

Comment:
I'm not sure this class is actually needed. 
I think you could just use: `mapping: dict[str, list[str]]`. This way you also don't need the `rename_mapping` property below since you can use `mapping` directly.

Reply:
I'd prefer follow the current implementation in the RegionProcessor, and then see how we can streamline or refactor to use common utils methods/classes...